analytics/js_tracking_api.rst

```diff
@@ -0,0 +1,924 @@
+    :returns: Undefined
+
+Commands
+--------
+
+
+Trigger tracking on demand
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+Trigger custom event
+````````````````````
+Trigger (custom) events bound to user actions::
+
+    _paq.push(["trackEvent", category, action, name, value, dimensions);
+
+.. data:: category
+
+    **Required** String with event category.
+
+.. data:: action
+
+    **Required** String with event action.
+
+.. data:: name
+
+    **Optional** String with event name.
+
+.. data:: value
+
+    **Optional** Number value with event value.
+
+.. data:: dimensions
+
+    **Optional**  :ref:`Custom dimension<Custom Dimensions>` that should be tracked with this action. Can be multiple dimensions.
+    Written as object property using ``dimension{ID}`` notation.
+
+    Example::
+
+        {
+           dimension1: "example value",
+           dimension2: "example value"
+        }
+
+Example of usage (tracking when user clicks on cancel button on exit intent)::
+
+        _paq.push(["trackEvent", "Exit intent", "Click on button", "Cancel"]);
+
+Track goal conversion
+`````````````````````
+Allows to manually track goal conversion Used in Goals - Days to Conversion report::
+
+    _paq.push(["trackGoal", goal_name, goal_value, dimensions]);
+
+.. data:: goal_name
+
+    **Required** String with Goal Name
+
+.. data:: goal_value
+
+    **Optional** Number value with tracked conversion value.
+
+.. data:: dimensions
+
+    **Optional**  :ref:`Custom dimension<Custom Dimensions>` that should be tracked with this action. Can be multiple dimensions.
+    Written as object property using ``dimension{ID}`` notation.
+
+    Example::
+
+        {
+           dimension1: "example value",
+           dimension2: "example value"
+        }
+
+Example of usage::
+
+    _paq.push(["trackGoal" 1, 15]);
+
+Ecommerce tracking
+^^^^^^^^^^^^^^^^^^
+
+Adding Ecommerce item
+`````````````````````
+To add ecommerce item (for example to track changes in users cart using ``trackEcommerceCartUpdate``) user ``addEcommerceItem`` function::
+
+    _paq.push(["addEcommerceItem", productSKU, productName, productCategory, productPrice, productQuantity]);
+
+.. data:: productSKU
+
+    **Required** String with product stock-keeping unit.
+
+.. data:: productName
+
+    **Optional** String with product name.
+
+.. data:: productCategory
+
+    **Optional** Product category, can be written as Array with up to 5 elements.
+
+.. data:: productPrice
+
+    **Optional** Float with product price.
+
+.. data:: productQuantity
+
+    **Optional** Float with product quantity.
+
+.. warning::
+
+    All string parameters should be escaped, without double quote character.
+
+.. warning::
+
+    You must do this each page view to maintain cart status.
+
+.. warning::
+
+    This function does not track any data to Tracker. This is helper function for ``trackEcommerceOrder`` or ``trackEcommerceCartUpdate`` function to populate items to send.
```

Comment:
I would put this warning before param description. I'm also not sure if it has to be warning. Maybe `note` would be enough.